<launch>
  <arg name="map_file" default="$(find gx2019_omni_simulations)/map/map_remote.yaml"/>

  <!-- open rplidar A2 -->
  <node name="rplidarNode" pkg="rplidar_ros" type="rplidarNode" output="screen">
    <param name="serial_port" type="string" value="/dev/rplidar"/>
    <param name="serial_baudrate" type="int" value="115200"/>
    <param name="frame_id" type="string" value="hokuyo_link"/>
    <param name="inverted" type="bool" value="false"/>
    <param name="angle_compensate" type="bool" value="true"/>
    <remap from="scan" to="my_robot/laser/scan"/>
  </node>
  <!-- open art imu -->
  <node pkg="art_imu" name="imu" type="art_imu" output="screen">
    <param name="port" value="/dev/ttyUSB0"/>
    <param name="frame_id" type="string" value="imu_link"/>
    <param name="model" value="art_imu_02a"/>
    <param name="baud" value="115200"/>
  </node>
  <!-- open usb camera -->
  <!-- <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device" value="/dev/video4" />
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="pixel_format" value="yuyv" />
    <param name="ca mera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node> -->

  <include file="$(find gx2019_omni_simulations)/launch/gx2019_localization.launch" />
  <param name="robot_description" command="cat '$(find gx2019_omni_simulations)/urdf/car.urdf'"/>
  <!-- Combine joint values -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>
  <!-- send fake joint values -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="use_gui" value="False"/>
  </node>

  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find gx2019_omni_simulations)/rviz/rviz_remote.rviz"/>

  <node pkg="tf" type="static_transform_publisher" name="camera" args="0.0 0.0 0.0 -1.57 0 -1.57 camera_link camera 100" />
</launch>
